src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/energy/EnergyNet.java

```diff
@@ -143,6 +143,13 @@ public void onClassificationChange(Location l, NetworkComponent from, NetworkCom
         }
     }
 
+    @Override
+    protected void addLocationToNetwork(@Nonnull Location l) {
```

Comment:
Why do we override it and not just fix the actual method in the networks class?

Reply:
I put it here because it doesn't make sense to have an energy network specific bug to be fixed in an abstract network class. An abstract class shouldn't call to its child class.

Reply:
Nothing else in Slimefun uses the network class except energy net tho. So I think it’s fine to add this check in there. I can be wrong tho

Reply:
Cargo net uses the network class I thought? Also like you said if an addon extended it then that would be another usage.

Reply:
Yep cargo net extends abstract item network which extends network

Reply:
Yes but if an addon uses this implementation it would also have the same issue

Reply:
Are there any addons extending Network and not EnergyNet? For an energy fix, it makes sense for it to be here. `Network` is a much more general-use class, not specific to energy.

Reply:
> Are there any addons extending Network and not EnergyNet for energy? For an energy fix, it makes sense for it to be here. `Network` is a much more general-use class, not specific to energy.

Personally I don't know of any